enhancements/agent-installer/host-based-mirror.md

```diff
@@ -0,0 +1,282 @@
+ImageContentSources section to install-config.yaml with the upstream sources
+and corresponding mirror locations. With this proposal, this step is no longer
+required, instead a tarfile containing the image sources must be included in
+the assets directory when running `openshift-install agent create image`.
+A new parameter flag `--embed-release-image=true` will be added to the `create
+image` command. When run in a connected environment, it will pull the images
+using the release image embedded in the installer. Alternatively, the user
+can generate this tarfile outside of the tool using `oc-mirror` to write to a
+file.  In either case, the ISO can be generated in a disconnected environment
+when the tarfile is available as it does not require access to upstream sources.
+
+Once the tarfile is provided, one of the challenges of this feature is how to
+include it in the ISO. A tarfile containing all of the container images for a
+release is on the order of 16GB. A separate ISO may be created to contain these
+images. The work that was done to create a separate partition for the
+[telco-ran-tools](https://github.com/openshift-kni/telco-ran-tools/blob/main/docs/partitioning.md)
+may be leveraged. In addition to the image tarfile, any tools needed during the
+host boot process (i.e. registry, `oc-mirror` etc.) must be included in the
+ignition when the ISO is created. This will utilize the work done for the
+[agent-tui](https://github.com/openshift/installer/pull/6777)
+to include binaries in compressed cpio archive appended to initrd.
+
+During host boot, the presence of the container image tarfile will indicate
+that a host-based mirror is to be used. The tarfile will be unpacked into
+container storage and `/etc/container/registries.conf` will be set so that
+all image downloads on the host, i.e. via `podman pull`, are made using this
+mirror. The bootstrap process will function in the usual manner with no
+external network required to run containers. No changes to pull secret or CA
+certificates will be needed to access this mirror.
+
+After the release image has been written to disk and bootstrap is complete, the
+container images must be made available upon reboot to complete the
+installation. It is envisioned that a method similar to the telco-ran-tools
+[ignition config override](https://github.com/openshift-kni/telco-ran-tools/blob/main/docs/ztp-precaching.md#43-nodesignitionconfigoverride)
+will be used. A script can be run using ignition config override to make the
+images available, create a registry in the cluster, and publish the images
+to the registry.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+To create the tarfile of container images from within the openshift-installer
+the best choice will be `oc-mirror`. The binary can be extracted from
+the release image, it's well supported, and well documented. When
+`openshift-install agent create container-image-file` is run a basic
```

Comment:
This command is really the same as `openshift-install agent create image --embed-release-image=true` it is intended to create the tarfile using the imageset info. Again this has been updated to remove the inclusion of the container-image tarfile and assume it will be available.